### PR TITLE
add iris and vtol models with cameras for avoidance and safe landing

### PR DESCRIPTION
**Pull request by mrivi**
This PR moves the models from the PX4 avoidance repository to SITL gazebo. The models already depends on the Firmware models so it made sense to move them here such that any modification to the plugin can be easily tested against them as well.

### CODE REVIEW

File: models/depth_camera/depth_camera.sdf [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+          <pointCloudTopicName>depth/points</pointCloudTopicName>
+          <frameName>camera_link</frameName>
+          <distortion_k1>0.0</distortion_k1>
+          <distortion_k2>0.0</distortion_k2>
+          <distortion_k3>0.0</distortion_k3>
+          <distortion_t1>0.0</distortion_t1>
+          <distortion_t2>0.0</distortion_t2>
+        </plugin>
+      </sensor>
+      <!-- <sensor name="laser" type="ray">
```
Comment (TSC21) [inline models/depth_camera/depth_camera.sdf]: Can we clean the commented elements here?

File: models/iris_depth_camera/iris_depth_camera.sdf [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<sdf version='1.5'>
```
Comment (TSC21) [inline models/iris_depth_camera/iris_depth_camera.sdf]: In order to keep the syntax highlights:
```suggestion
<?xml version="1.0" ?>
<sdf version="1.5">
```

File: models/iris_depth_camera/model.config [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0"?>
+<model>
+  <name>3DR Iris with depth camera</name>
+  <version>1.0</version>
+  <sdf version='1.4'>iris_depth_camera.sdf</sdf>
```
Comment (TSC21) [inline models/iris_depth_camera/model.config]: You state in the `iris_depth_camera.sdf` that the SDF version is 1.5 and here that is 1.4. Please change it to 1.5 so to keep it consistent.

File: models/iris_depth_camera/model.config [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+  <author>
+   <name>Lorenz Meier and Thomas Gubler</name>
+   <email>lorenz@px4.io</email>
+  </author>
+
```
Comment (TSC21) [inline models/iris_depth_camera/model.config:11]: Maybe you want to add you here as the author as well?

File: models/iris_depth_camera_3/iris_depth_camera_3.sdf [hunk 5]
```diff
@@ -0,0 +1,237 @@
+<sdf version='1.5'>
```
Comment (TSC21) [inline models/iris_depth_camera_3/iris_depth_camera_3.sdf]:
```suggestion
<?xml version="1.0" ?>
<sdf version="1.5">
```

File: models/iris_depth_camera_3/iris_depth_camera_3.sdf [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+  <model name='iris_depth_camera_3'>
```
Comment (TSC21) [inline models/iris_depth_camera_3/iris_depth_camera_3.sdf]: In terms of naming: can we try to differentiate between `iris_depth_camera` and `iris_depth_camera_3` without using numbering? Like a specific thing that one has and the other doesn't? Having a `iris_depth_camera` and `iris_depth_camera_3` listed doesn't seem adequate.

File: models/iris_depth_camera_3/iris_depth_camera_3.sdf [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+        <limit>
+          <upper>0</upper>
+          <lower>0</lower>
+        </limit>
+      </axis>
+    </joint>
+
+    <include>
+      <uri>model://lidar</uri>
+      <pose>-0.12 0 0 0 3.1415 0</pose>
```
Comment (TSC21) [inline models/iris_depth_camera_3/iris_depth_camera_3.sdf]: This has to have zero rotation in pitch and roll in order to have the sensor facing downwards:
```suggestion
      <pose>-0.12 0 0 0 0 0</pose>
```

File: models/iris_depth_camera_3/model.config [hunk 8]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0"?>
+<model>
+  <name>3DR Iris with 3 depth cmaera</name>
+  <version>1.0</version>
+  <sdf version='1.4'>iris_depth_camera_3.sdf</sdf>
```
Comment (TSC21) [inline models/iris_depth_camera_3/model.config]:
```suggestion
  <sdf version='1.5'>iris_depth_camera_3.sdf</sdf>
```

File: models/iris_depth_camera_3/model.config [hunk 9]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0"?>
+<model>
+  <name>3DR Iris with 3 depth cmaera</name>
```
Comment (TSC21) [inline models/iris_depth_camera_3/model.config]:
```suggestion
  <name>3DR Iris with 3 depth camera</name>
```

File: models/iris_depth_camera_3/model.config [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+  <version>1.0</version>
+  <sdf version='1.4'>iris_depth_camera_3.sdf</sdf>
+  <author>
+   <name>Lorenz Meier and Thomas Gubler</name>
+   <email>lorenz@px4.io</email>
+  </author>
+  <description>
+    This is a model of the 3DR Iris Quadrotor with 3 depth cameras.
```
Comment (TSC21) [inline models/iris_depth_camera_3/model.config]: Ok so the main difference between `iris_depth_camera` and `iris_depth_camera_3` are the number of cameras? What about `iris_triple_depth_camera`? Seems more suited than `iris_depth_camera_3`, as it the `_3` suffix seems like it's version 3 of `iris_depth_camera`.

File: models/iris_downwards_camera/iris_downwards_camera.sdf [hunk 11]
```diff
@@ -0,0 +1,25 @@
+<sdf version='1.5'>
```
Comment (TSC21) [inline models/iris_downwards_camera/iris_downwards_camera.sdf]:
```suggestion
<?xml version="1.0" ?>
<sdf version="1.5">
```

File: models/iris_downwards_camera/model.config [hunk 12]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0"?>
+<model>
+  <name>3DR Iris</name>
+  <version>1.0</version>
+  <sdf version='1.4'>iris_downwards_camera.sdf</sdf>
```
Comment (TSC21) [inline models/iris_downwards_camera/model.config]:
```suggestion
  <sdf version='1.5'>iris_downwards_camera.sdf</sdf>
```

File: models/stereo_camera/model.config [hunk 13]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0"?>
+
```
Comment (TSC21) [inline models/stereo_camera/model.config]:
```suggestion
```

File: models/stereo_camera/stereo_camera.sdf [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+          <!-- <rightFrameName>camera_link</rightFrameName> -->
+          <hackBaseline>0.07</hackBaseline>
+          <distortionK1>0.0</distortionK1>
+          <distortionK2>0.0</distortionK2>
+          <distortionK3>0.0</distortionK3>
+          <distortionT1>0.0</distortionT1>
+          <distortionT2>0.0</distortionT2>
+        </plugin>
+      </sensor>
+    </link>
```
Comment (TSC21) [inline models/stereo_camera/stereo_camera.sdf:79]: Does it make sense to add a downward facing lidar here as well?

File: models/iris_stereo_camera/iris_stereo_camera.sdf [hunk 15]
```diff
@@ -0,0 +1,24 @@
+<sdf version='1.5'>
```
Comment (TSC21) [inline models/iris_stereo_camera/iris_stereo_camera.sdf:2]:
```suggestion
<?xml version="1.0" ?>
<sdf version="1.5">
```

File: models/iris_stereo_camera/model.config [hunk 16]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0"?>
+<model>
+  <name>3DR Iris with stereo camera</name>
+  <version>1.0</version>
+  <sdf version='1.4'>iris_stereo_camera.sdf</sdf>
```
Comment (TSC21) [inline models/iris_stereo_camera/model.config]:
```suggestion
  <sdf version='1.5'>iris_stereo_camera.sdf</sdf>
```

File: models/vtol_downwards_camera/vtol_downwards_camera.sdf [hunk 17]
```diff
@@ -0,0 +1,24 @@
+<sdf version='1.5'>
```
Comment (TSC21) [inline models/vtol_downwards_camera/vtol_downwards_camera.sdf]:
```suggestion
<?xml version="1.0"?>
<sdf version='1.5'>
```

File: models/vtol_downwards_camera/model.config [hunk 18]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0"?>
+<model>
+  <name>Standard VTOL with downward facing camera</name>
+  <version>1.0</version>
+  <sdf version='1.4'>vtol_downwards_camera.sdf</sdf>
```
Comment (TSC21) [inline models/vtol_downwards_camera/model.config]:
```suggestion
  <sdf version='1.5'>vtol_downwards_camera.sdf</sdf>
```

File: models/vtol_downwards_camera/model.config [hunk 19]
```diff
@@ -0,0 +1,14 @@
...
+
+  <author>
+   <name>Lorenz Meier and Thomas Gubler</name>
+   <email>lorenz@px4.io</email>
+  </author>
+
+  <description>
```
Comment (TSC21) [inline models/vtol_downwards_camera/model.config]: Can you add a description here as well?

File: models/stereo_camera/meshes/hokuyo.dae [hunk 20]
```diff
@@ -0,0 +1,253 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (TSC21) [inline models/stereo_camera/meshes/hokuyo.dae:1]: The `hokuyo.dae` meshes get repeated in the `stereo_camera` and the `depth_camera`. Is there a way we can have a specific model for each of the cameras? Example - Realsense camera: https://github.com/intel/gazebo-realsense/tree/master/models/realsense_camera

**Review by TSC21 — CHANGES_REQUESTED** [review]
Just a thing here and there. Otherwise looks good.

File: models/vtol_downwards_camera/vtol_downwards_camera.sdf [hunk 21]
```diff
@@ -0,0 +1,24 @@
...
+  <model name='vtol_downwards_camera'>
```
Comment (Jaeyoung-Lim) [inline models/vtol_downwards_camera/vtol_downwards_camera.sdf]: The notion of camera is confusing as there are camera models in this repo that uses the `gst-camera-plugin`, which is a normal camera, not a depth camera. Could we possibly have a better name representing depth cameras?

File: models/iris_depth_camera_3/iris_depth_camera_3.sdf (hunk 6 above)
Comment (mrivi) [reply models/iris_depth_camera_3/iris_depth_camera_3.sdf]: The differentiation come from the number of cameras, hence the 3.

Comment (TSC21) [reply models/iris_depth_camera_3/iris_depth_camera_3.sdf]: Yes. Check: https://github.com/PX4/sitl_gazebo/pull/358#discussion_r335365125

File: models/vtol_downwards_camera/vtol_downwards_camera.sdf (hunk 21 above)
Comment (mrivi) [reply models/vtol_downwards_camera/vtol_downwards_camera.sdf]: @Jaeyoung-Lim  do you have a naming suggestion?

Comment (Jaeyoung-Lim) [reply models/vtol_downwards_camera/vtol_downwards_camera.sdf]: May be `vtol_depth_camera`? This is also consistent with the `iris_depth_camera`. I also realized this is also true for the `iris_downward_camera`

**Comment by TSC21** [comment]
@mrivi the schema validation is complaining about the usage of the model inside another: https://travis-ci.org/PX4/sitl_gazebo/jobs/598558613#L282. This should be fixed if you update the SDF version to 1.5 or 1.6. Though, another ways is to use the `<include></include>` elements, similar to what you have in `iris_depth_camera`. You can basically do the same for the 3 cameras by using the `<include></include>` and then set the name and pose of each inside it. Example: http://gazebosim.org/tutorials?tut=add_laser

**Review by TSC21 — CHANGES_REQUESTED** [review]
@mrivi on the `iris_triple_depth_camera` please use `include` instead of `model`, so we can have the schema validator passing. Looks like updating to 1.5 didn't help. Also please add `<?xml version="1.0" ?>` to it.

**Comment by mrivi** [comment]
Triple cam iris with the realsense mesh
iris_3_cam

**Comment by Jaeyoung-Lim** [comment]
Looks awesome!

**Review by TSC21 — APPROVED** [review]
Good work!

**Pull request merged**